src/main/java/duke/io/Storage.java

```diff
@@ -0,0 +1,108 @@
+package duke.io;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.Task;
+import duke.task.TaskList;
+
+/**
+ * Handles all file saving and loading operations of Duke. <br>
+ */
+public class Storage {
+    /**
+     * Saves the content of the provided Task ArrayList into the file specified. <br>
+     * Tasks are delimited by ';'. <br>
+     * Tasks fields are delimited by ',', in the following format: <br>
+     * <pre> TodoTask Task: T ,Task Done Status ,Task Name. </pre>
+     * <pre> DeadlineTask Task: D, Task Done Status ,Task Name, Task Deadline Date. </pre>
+     * <pre> EventTask: E ,Task Done Status ,Task Name, Task Event Date. <br> </pre>
+     *
+     * @param folderName folder name to create the save file.
+     * @param fileName name of save file.
+     * @param arr list of tasks to be saved.
+     * @return 0: save success, -1: error encountered.
+     */
+    public static int saveFile(String folderName, String fileName, ArrayList<Task> arr) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/M/yyyy");
+        try {
+            Files.createDirectories(Paths.get(folderName));
+            String filePath = folderName + "/" + fileName;
+            File myObj = new File(filePath);
```

Comment:
Consider changing `myObj` to a more descriptive variable name.
```suggestion
            File savedFile = new File(filePath);
```